### ISSUE TITLE
Can ROOT-5 be compiled with GCC-6.3.0 ?

### ISSUE THREAD

**Issue by brucellino** [SouthAfricaDigitalScience/root-deploy#67]
In local testing, I'm getting some gcc errors : 

```
[ 37%] Generating G__Reflex.cxx, G__Reflex.h
Error: class,struct,union or type decltype(nullptr) not defined  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/x86_64-pc-linux-gnu/bits/c++config.h:205:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:162:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:162:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:163:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:163:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:164:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:165:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:166:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:166:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:181:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:198:
Error: Symbol bool is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:244:
Error: Function alignof(bool) is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:244:
Error: Invalid type 'constexpr' in declaration of 'int _S_alignment' /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:244:
Syntax error /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:246:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:246:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:249:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:249:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:250:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:250:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:251:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:252:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:253:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:253:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:380:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:401:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:420:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:462:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:506:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:519:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:529:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:539:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:549:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/bits/atomic_base.h:559:
Error: abstract class object '__atomic_base<bool> _M_base' declared /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:61:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:64:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:64:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:65:
Error: Symbol default is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:65:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:66:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:67:
Error: unrecognized language construct /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:68:
Error: Symbol delete is not defined in current scope  /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:68:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:98:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:106:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:115:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:135:
Error: Redefinition of default argument /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:155:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Missing one of ' 	
;:=+-)(*&^%$#@!~'"\|][}{/?.>,<' expected at or after line 42865.
Error: Unexpected end of file (G__fgetstream():2) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
Error: Unexpected end of file (G__fgetc()) /data/ci-build/generic/centos6/x86_64/gcc/6.3.0/include/c++/6.3.0/atomic:42865:
make[2]: *** [cint/reflex/G__Reflex.cxx] Error 1
make[2]: *** Deleting file `cint/reflex/G__Reflex.cxx'
make[1]: *** [cint/reflex/CMakeFiles/ReflexDict.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by bazinski** [comment]
I would say no.
root5 with 4.9.4 and no others

**Comment by brucellino** [comment]
Closed in 07f881d. ROOT builds are separated into separate jobs.

**Comment by brucellino** [comment]
This issue was moved to http://discourse.sci-gaia.eu/t/can-root-5-be-compiled-with-gcc-6-3-0/2617
